### ISSUE TITLE
Fails to build on Windows

### ISSUE THREAD

**Issue by veelo** [funkwerk-mobility/serialized#2]
The problem is Windows not having a case sensitive file system and `serialized` not following [best practices from the spec](https://dlang.org/spec/module.html) section 4.1.5 and 4.1.8.

```
D:\SARC\xmltest>dub build
    Starting Performing "debug" build using C:\D\dmd2\windows\bin64\dmd.exe for x86_64.
  Up-to-date boilerplate 1.9.1: target for configuration [library] is up to date.
  Up-to-date prettyprint 1.0.9: target for configuration [library] is up to date.
  Up-to-date unit-threaded:from 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:exception 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:assertions 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:runner 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:behave 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:integration 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:mocks 2.1.6: target for configuration [library] is up to date.
  Up-to-date unit-threaded:property 2.1.6: target for configuration [default] is up to date.
  Up-to-date unit-threaded 2.1.6: target for configuration [library] is up to date.
  Up-to-date dshould 1.7.1: target for configuration [library] is up to date.
  Up-to-date dxml 0.4.3: target for configuration [library] is up to date.
    Building serialized 1.11.0: building configuration [library]
C:\Users\bastiaan\AppData\Local\dub\packages\serialized-1.11.0\serialized\src\text\xml\Validation.d(12,8): Error: module `text.xml.tree` from file C:\Users\bastiaan\AppData\Local\dub\packages\serialized-1.11.0\serialized\src\text\xml\Tree.d must be imported with 'import text.xml.tree;'
C:\Users\bastiaan\AppData\Local\dub\packages\serialized-1.11.0\serialized\src\text\xml\Decode.d(10,8): Deprecation: module `text.xml.tree` is deprecated - Please use text.xml.Tree instead
C:\Users\bastiaan\AppData\Local\dub\packages\serialized-1.11.0\serialized\src\text\xml\Parser.d(10,8): Deprecation: module `text.xml.tree` is deprecated - Please use text.xml.Tree instead
C:\Users\bastiaan\AppData\Local\dub\packages\serialized-1.11.0\serialized\src\text\xml\Writer.d(8,8): Deprecation: module `text.xml.tree` is deprecated - Please use text.xml.Tree instead
Error C:\D\dmd2\windows\bin64\dmd.exe failed with exit code 1.

D:\SARC\xmltest>dmd --version
DMD64 D Compiler v2.102.2-dirty
Copyright (C) 1999-2023 by The D Language Foundation, All Rights Reserved written by Walter Bright
```

On closer inspection, Git on Windows failed to properly clone the `serialized` repository due to `Tree.d` and `tree.d` conflicting, and `Tree.d` actually getting the contents of the original `tree.d`! These are the contents of `Tree.d` as checked out on Windows:
```d
deprecated("Please use text.xml.Tree instead")
module text.xml.tree;

public import text.xml.Tree;
```
This module deprecated itself and then recursively imports itself (on Windows). The contents of https://github.com/funkwerk-mobility/serialized/blob/master/src/text/xml/Tree.d are not present in my checkout.

**Comment by FeepingCreature** [comment]
I've checked internally and it looks like pretty much nobody is using the old import anymore. Removed, fixed 1.12.0.

Thanks for the report! Feel free to keep 'em coming.

**Comment by veelo** [comment]
Thank you!
